### PR TITLE
Feature/#73030.warmup service

### PR DESCRIPTION
**Pull request by yevgenyka**

### CODE REVIEW

File: Gigya.Microdot.Hosting/HttpService/Endpoints/HealthEndpoint.cs [hunk 1]
```diff
@@ -37,12 +37,15 @@ public class HealthEndpoint : ICustomEndpoint
...
-        public HealthEndpoint(IServiceEndPointDefinition serviceEndPointDefinition, IServiceInterfaceMapper serviceInterfaceMapper, IActivator activator,IServiceDrainListener drainListener)
+        private readonly IWarmup _warmup;
+
+        public HealthEndpoint(IServiceEndPointDefinition serviceEndPointDefinition, IServiceInterfaceMapper serviceInterfaceMapper, IActivator activator,IServiceDrainListener drainListener, IWarmup warmap)
         {
             _drainListener = drainListener;
             ServiceEndPointDefinition = serviceEndPointDefinition;
             ServiceInterfaceMapper = serviceInterfaceMapper;
             Activator = activator;
+            _warmup = warmap;
```
Comment (bronsh) [inline Gigya.Microdot.Hosting/HttpService/Endpoints/HealthEndpoint.cs]: remove

File: tests/Gigya.Microdot.Orleans.Hosting.UnitTests/WarmupTests.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            sw.Stop();
+
+            Assert.Less(sw.Elapsed.TotalMilliseconds, AssemblyInitialize.ResolutionRoot.Get<DependantClass>().SleepTime);
+            Assert.AreEqual(result, 2);
+
+            tester.Dispose();
+        }
+
+        [Test]
+        public async Task NoWarmup_InstanceNotReadyBeforeCallingMethod()
```
Comment (bronsh) [inline tests/Gigya.Microdot.Orleans.Hosting.UnitTests/WarmupTests.cs]: remove this test

File: Gigya.Microdot.Orleans.Ninject.Host/MicrodotOrleansServiceHost.cs [hunk 3]
```diff
@@ -75,6 +76,8 @@ protected override void OnStart()
             PreInitialize(Kernel);
             OnInitilize(Kernel);
 
+            Warmup(Kernel);
+
             SiloHost = Kernel.Get<GigyaSiloHost>();
```
Comment (bronsh) [inline Gigya.Microdot.Orleans.Ninject.Host/MicrodotOrleansServiceHost.cs]: Add test that validates Warmup occurs **before** service starts its listener

File: Gigya.Microdot.Orleans.Hosting/OrleansServiceInterfaceMapper.cs [hunk 4]
```diff
@@ -35,7 +35,7 @@ namespace Gigya.Microdot.Orleans.Hosting
     public class OrleansServiceInterfaceMapper : ServiceInterfaceMapper
     {
         public override IEnumerable<Type> ServiceInterfaceTypes => Mappings.Keys;
-        public IEnumerable<Type> ServiceClassesTypes { get; }
+        public override IEnumerable<Type> ServiceClassesTypes { get; }
```
[the remark this answers is not in the document]
Comment (yevgenyka) [reply Gigya.Microdot.Orleans.Hosting/OrleansServiceInterfaceMapper.cs]: Instead of overriding, the protected set was added to the base property

**Pull request merged**